### DISCUSSION TITLE
Can't set non-/64 mesh-local prefixes with ot-cli

### DISCUSSION THREAD

**Discussion by spacekitteh** [openthread/openthread#11083] [General]
**Describe the bug** A clear and concise description of what the bug is.
As far as I'm aware, mesh-local prefixes aren't required to be /64s. When trying to set a non-/64 prefix, ot-cli fails to parse it. 

E.g.:

```
> dataset meshlocalprefix fd00:db8:3::/48    
Error 6: Parse
> dataset meshlocalprefix [fd00:db8::]/16
Error 6: Parse
```

**Comment by abtink** [comment]
The mesh-local prefix should be 64 bits long, as it is used to construct RLOC, ALOC, and ML-EID addresses. Section 8.10.1.8 (in the Thread specification), where the "Network Mesh-Local Prefix TLV" is defined, explicitly states "length = 8".

The CLI command `dataset meshlocalprefix` does not require the prefix length (`/64`) to be specified when setting the mesh-local prefix. Here are links to its documentation, which provide examples:

- [README_DATASET.md#meshlocalprefix](https://github.com/openthread/openthread/blob/143ebbfb7c32d0ba2a85ce58689be980a2cb92e7/src/cli/README_DATASET.md#meshlocalprefix)
- [openthread.io CLI documentation](https://openthread.io/reference/cli/commands#dataset_meshlocalprefix_getset)
    
---

This is also reflected in the code and definition of `otMeshLocalPrefix`:

```c++
#define OT_MESH_LOCAL_PREFIX_SIZE OT_IP6_PREFIX_SIZE ///< Size of the Mesh Local Prefix (bytes)

/**
 * Represents a Mesh Local Prefix.
 */
typedef otIp6NetworkPrefix otMeshLocalPrefix;
```

and

```c++
#define OT_IP6_PREFIX_SIZE 8                            ///< Size of an IPv6 prefix (bytes)
#define OT_IP6_PREFIX_BITSIZE (OT_IP6_PREFIX_SIZE * 8)  ///< Size of an IPv6 prefix (bits)
```

**Comment by jwhui** [comment]
[RFC 4944 Section 6](https://datatracker.ietf.org/doc/html/rfc4944#section-6) states:

```
An IPv6 address prefix used for stateless autoconfiguration RFC4862
of an IEEE 802.15.4 interface MUST have a length of 64 bits.
```
